test/model/deepvar_hierarchical/generate_hierarchical_dataset.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+Loads the datasets used in Salinas et al. 2019 (https://tinyurl.com/woyhhqy).
+This wrapper downloads and unpacks them so they don'thave to be attached as
+large files in GluonTS master.
+"""
+
+from typing import NamedTuple
+import numpy as np
+import pandas as pd
+
+from gluonts.dataset.common import ListDataset
+
+
+class HierarchicalMetaData(NamedTuple):
+    S: np.ndarray
+    freq: str
+
+
+class HierarchicalTrainDatasets(NamedTuple):
+    train: ListDataset
+    test: ListDataset
+    metadata: HierarchicalMetaData
+
+
+def sine7(seq_length: int, prediction_length: int):
```

Comment:
~There’s no test function in this script, is this script needed?~
Disregard this, my browser doesn’t allow me to delete this comment for some reason